### PR TITLE
[kf5bluezqt] Add documentation package. Fixes JB#58716

### PR DESCRIPTION
**Pull request by vigejolla**

### CODE REVIEW

File: rpm/kf5bluezqt-bluez5.spec [hunk 1]
```diff
@@ -63,6 +71,10 @@ This package contains the development header files for kf5bluezqt
 %install
 %make_install
 
+doxygen ../doc/Doxyfile
+mkdir -p %{buildroot}/%{_docdir}/%{name}
+install -m 644 -D -t %{buildroot}/%{_docdir}/%{name} doc/html/*.html doc/html/*.png doc/html/*.css
```
Comment (martyone) [inline rpm/kf5bluezqt-bluez5.spec]: Do we actually exclude here something? If so, could it be done more explicitly, e.g. by removing what we exclude before we copy the whole html dir?

**Review by martyone — APPROVED** [review]
One suggestion, otherwise LGTM

File: rpm/kf5bluezqt-bluez5.spec (hunk 1 above)
Comment (vigejolla) [reply rpm/kf5bluezqt-bluez5.spec]: I'm sure we do, otherwise I wouldn't have done it like. I'll have a look.

Comment (vigejolla) [reply rpm/kf5bluezqt-bluez5.spec]: Turns out we were excluding a bit too much. So, a second attempt is now ready for review - this time including the search directory. And the only file which is excluded on purpose is removed like you suggested.

Comment (martyone) [reply rpm/kf5bluezqt-bluez5.spec]: nice :)

File: rpm/kf5bluezqt-bluez5.spec [hunk 2]
```diff
@@ -63,6 +71,13 @@ This package contains the development header files for kf5bluezqt
 %install
 %make_install
 
+doxygen ../doc/Doxyfile
+mkdir -p %{buildroot}/%{_docdir}/%{name}
+mkdir -p %{buildroot}/%{_docdir}/%{name}/search
+rm doc/html/installdox
+install -m 644 -D -t %{buildroot}/%{_docdir}/%{name} doc/html/* || true
+install -m 644 -D -t %{buildroot}/%{_docdir}/%{name}/search doc/html/search/*
```
Comment (martyone) [inline rpm/kf5bluezqt-bluez5.spec]: Maybe use plain `cp -r`? The permissions would likely be correct already on the source side anyway.

**Comment by martyone** [comment]
One more idea: could we also build qch from the start?

File: rpm/kf5bluezqt-bluez5.spec (hunk 2 above)
Comment (vigejolla) [reply rpm/kf5bluezqt-bluez5.spec]: The permissions indeed seem to be correct; changed to `cp -r`

File: rpm/kf5bluezqt-bluez5.spec [hunk 3]
```diff
@@ -53,6 +54,13 @@ Requires:   %{name} = %{version}
 %description devel
 This package contains the development header files for kf5bluezqt
 
+%package doc
+Summary:    API documentation for %{name}
+BuildArch:  noarch
```
Comment (Thaodan) [inline rpm/kf5bluezqt-bluez5.spec:59]: Is no BuildArch here needed? This is the first package of this kind where I see it.

Comment (vigejolla) [reply rpm/kf5bluezqt-bluez5.spec:59]: It's pretty much in all of our documentation packages.

File: rpm/kf5bluezqt-bluez5.spec [hunk 4]
```diff
@@ -63,6 +71,12 @@ This package contains the development header files for kf5bluezqt
 %install
 %make_install
 
+doxygen ../doc/Doxyfile
```
Comment (Thaodan) [inline rpm/kf5bluezqt-bluez5.spec:74]: This should be unneeded.
Cmake can generate the api docs.

**Comment by Thaodan** [comment]
> One more idea: could we also build qch from the start?

That should be possible: https://invent.kde.org/frameworks/bluez-qt/-/blob/v5.50.0/CMakeLists.txt#L66

**Comment by vigejolla** [comment]
> > One more idea: could we also build qch from the start?
> That should be possible: https://invent.kde.org/frameworks/bluez-qt/-/blob/v5.50.0/CMakeLists.txt#L66

Good idea, but I think it's out of scope for this task. I created a separate task for this.

**Pull request merged**